dts/bindings/pinctrl/nxp,mcux-rt-pinctrl.yaml

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) 2022 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  The node has the 'pinctrl' node label set in MCUX RT SoC's devicetree. These
+  nodes can be autogenerated using the MCUXpresso config tools combined with
+  the rt_dts_gen.py script in NXP's HAL. The mux, mode, input, daisy, and cfg
+  fields in a group select the pins to be configured, and the remaining
+  devicetree properties set configuration values for those pins
+  for example, here is an group configuring LPUART1 pins:
+
+  group0 {
+    pins = <&iomuxc_gpio_ad_b0_12_lpuart1_tx,
```

Comment:
last nitpick: `pinmux`

Reply:
Fixed in latest push. Thank you for working through this PR with me, I know it's taken a lot of your time but I think the code has improved immensely since the initial revision, and I'm really happy with the final result.